src/sys/unix.rs

```diff
@@ -1018,6 +1053,12 @@ impl Socket {
     }
 }
 
+impl Drop for Socket {
```

Comment:
Revert this; now `crate::Socket` is closed twice.

Reply:
It's not though. `crate::Socket` doesn't contain a `sys::Socket`. And `sys::Socket` implements `IntoRawFd`, so it should own it's file descriptor. Or it should not implement `IntoRawFd`.

From the docs of `IntoRawFd`:
> A trait to express the ability to consume an object and acquire ownership of its raw file descriptor.

If `sys::Socket` doesn't own it's filedescriptor, it can't make that promise.